swagger.yaml

```diff
@@ -75,12 +192,11 @@ paths:
           schema:
             $ref: '#/definitions/Datasets'
         500:
-          description: "Failed to process the request due to an internal error"
-
+          $ref: '#/responses/InternalError'
   /datasets/{id}:
     get:
       tags:
-      - "Dataset"
+      - "Public"
```

Comment:
done

Reply:
So the scenario you mention is true and actually the data structure returned is actually different for an internal user and the public for this very reason. The spec only shows/describes the model for the public user NOT a private user <- the reason for this is because you can't have a separate model for the same endpoint in swagger to describe the two scenarios (or at least i could not find how to do this). The other option is we change the spec to show the structure returned for a private user and describe fields so public users know what to get back.

I would argue at this point the private endpoints should not be displayed to the public at all (via a developer site unless you are authenticated, which might mean we need to figure out how to describe two models for the same endpoint?)